spec/ruby/language/method_spec.rb

```diff
@@ -176,6 +176,22 @@ def m(x)
       m('a', &nil).should == [false]
     end
   end
+
+  context "with a replaced singleton class" do
```

Comment:
language is not a great place for this, especially it shouldn't `require "socket"`, I'll move this to IO#reopen, it's really an edge case caused by IO#reopen